compiler/rustc_parse_format/src/lib.rs

```diff
@@ -986,6 +1009,19 @@ fn find_width_map_from_snippet(
     InputStringKind::Literal { width_mappings }
 }
 
+fn unescape_string(string: &str) -> Result<string::String, unescape::EscapeError> {
+    let mut buf = string::String::new();
+    let mut error = Ok(());
```

Comment:
```suggestion
    let mut ok = true;
```
and `ok.then(|| buf)` in the end.